lib/VerifiableCredentialStore.js

```diff
@@ -3,5 +3,161 @@
+  }
+
+  /**
+   * Gets a verifiable credential by its ID
+   *
+   * @param {string} id
+   */
+  async get({id}) {
+    const [doc] = await this.hub.find({equals: {id}});
+    if(!doc) {
+      const err = new Error('Verifiable Credential not found.');
+      err.name = 'NotFoundError';
+      throw err;
+    }
+    return doc.content;
+  }
+
+  /**
+   * Gets all verifiable credential instances that match the given parameters
+   *
+   * @param {string} [type]
+   * @param {string} [issuer]
+   *
+   * @return {Promise<Array>} List of matching VCs
+   */
+  async find({type, issuer}) {
+    const equals = [];
+    if(type) {
+      if(type instanceof Array) {
+        const query = type.map(type => ({type}));
+        equals.push(...query);
+      } else {
+        equals.push({type});
+      }
+    }
+    if(issuer) {
+      equals.push({issuer});
+    }
+    const results = await this.hub.find({equals});
+    return results.map(({content}) => content);
+  }
+
+  async _queryByExample({credentialQuery}) {
+    if(!credentialQuery) {
+      throw new Error('"credentialQuery" is needed to execute a QueryByExample.');
+    }
+    if(typeof credentialQuery !== 'object') {
+      throw new Error('"credentialQuery" must be an object or an array.');
+    }
+  
+    // normalize query to be an array
+    let query;  
+    if(credentialQuery instanceof Array) {
+      query = credentialQuery;
+    } else {
+      query = [credentialQuery];
+    }
+
+    const _query = async ({example}) => {
+      const {type, trustedIssuer = []} = example;
+      // normalize trusted issuers to be an array
+      let trustedIssuers;
+      if(trustedIssuer instanceof Array) {
+        trustedIssuers = trustedIssuer;
+      } else {
+        trustedIssuers = [trustedIssuer];
+      }
+      const issuer = trustedIssuers.filter(({required}) => required)
+        .map(({issuer}) => issuer);
+      const results = await this.find({issuer, type});
+      if(results.length === 0) {
+        const err = new Error(`Unable to find required credential.`);
+        err.name = 'NotFoundError';
+        err.query = example;
+        throw err;
+      }
+      return results
+    };
+
+    // // only look for credentials that are required
+    // const requiredQuery =  query.filter(({required}) => required);
+    // const requiredCredentials = await Promise.all(requiredQuery.map(_query));
+    const requiredCredentials = await Promise.all(query.map(_query));
+
+    // flatten results
+    const credentials = requiredCredentials
+      .reduce((acc, val) => acc.concat(val), []);
+    return credentials;
+  }
+  
+  /**
+   * Finds the best matching verifiable credential for the given query
+   *
+   * @param query a VerifiablePresentation credential handler query
+   *   (e.g. right now, support `QueryByExample`).
+   * @param engine
+   *
+   * @return {Promise<Array>} List of matching VCs
+   */
+  async match({query, engine = credentials => credentials}) {
```

Comment:
The default here makes no sense... should it be?

```suggestion
  async match({query, engine = credentialQuery => []}) {
```